<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="0.6" />
  <arg name="ray_length" default="5" />

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />
  <arg name="world_frame" default="map" />

  <group ns="$(arg UAV_NAME)">
    <node pkg="motion_planning_real_world" type="AEPReal" name="planner_node" output="screen">
      <rosparam file="$(find rrt_construction)/config/GainConfig.yaml" />
      <rosparam file="$(find motion_planning_real_world)/config/AEPlannerReal.yaml" />
      <param name="uav_namespace" value="$(arg UAV_NAME)" />
      <!-- Voxblox Planner -->
      <remap from="~esdf_map_in" to="voxblox_node/esdf_map_out" />
      <remap from="~tsdf_map_in" to="voxblox_node/tsdf_map_out" />

      <param name="use_tf_transforms" value="false" />
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="update_mesh_every_n_sec" value="0.0" />
      <param name="publish_traversable" value="false" />

      <!-- Publishers -->
      <remap from="~visualization_marker_out" to="visualization_marker_out" />
      <remap from="~simulation_ready" to="simulation_ready" />
      <remap from="~tree_node_out" to="tree_node" />
      <remap from="~frustum_out" to="frustum_out" />
      <remap from="~unknown_voxels_out" to="unknown_voxels_out" />
      <remap from="~offset_out" to="offset_out" />
      <!-- Subscribers -->
      <remap from="~local_pose_in" to="/mavros/local_position/pose" />
      <!-- Services clients-->
      <remap from="~setpoint_out" to="/mavros/setpoint_raw/local" />
      <remap from="~best_node_out" to="cache_nodes/best_node" />
      <remap from="~gp_query_out" to="cache_nodes/gp_query" />
      <!-- Services servers-->
      <remap from="~start_in" to="~start" />
      <remap from="~stop_in" to="~stop" />
      <remap from="~offset_in" to="~offset" />

    </node>

  </group>

</launch>
